teamcity: failed tests: TestReplicateRemovedNodeDisruptiveElection

The following tests appear to have failed:

[#899194](https://teamcity.cockroachdb.com/viewLog.html?buildId=899194):


```
--- FAIL: test/TestReplicateRemovedNodeDisruptiveElection (5.520s)
client_raft_test.go:3197: did not get expected error
------- Stdout: -------
I180911 21:58:09.664432 70253 gossip/gossip.go:408  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42405" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180911 21:58:09.666810 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n1 [sentinel];
 
W180911 21:58:09.671423 70253 gossip/gossip.go:1518  [n2] no incoming or outgoing connections
I180911 21:58:09.671803 70627 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:42405
I180911 21:58:09.674163 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n1 [sentinel];
 
I180911 21:58:09.674258 70253 gossip/gossip.go:408  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43957" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
W180911 21:58:09.696453 70253 gossip/gossip.go:1518  [n3] no incoming or outgoing connections
I180911 21:58:09.696551 70253 gossip/gossip.go:408  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:46753" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180911 21:58:09.697558 70758 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:42405
I180911 21:58:09.697702 70753 gossip/gossip.go:942  [n3] gossip connectivity
 
I180911 21:58:09.698267 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n1 [sentinel];
  n2 -> n1;
I180911 21:58:09.698772 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n1 [sentinel];
  n2 -> n1;
I180911 21:58:09.698995 70753 gossip/gossip.go:942  [n3] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I180911 21:58:09.699183 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I180911 21:58:09.699267 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
W180911 21:58:09.775366 70253 gossip/gossip.go:1518  [n4] no incoming or outgoing connections
I180911 21:58:09.775494 70253 gossip/gossip.go:408  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:37093" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180911 21:58:09.776520 70309 gossip/client.go:129  [n4] started gossip client to 127.0.0.1:42405
I180911 21:58:09.776971 70872 gossip/gossip.go:942  [n4] gossip connectivity
 
I180911 21:58:09.780151 70872 gossip/gossip.go:942  [n4] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:09.781194 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:09.781748 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:09.781956 70753 gossip/gossip.go:942  [n3] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:09.837164 70253 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n2 established
I180911 21:58:09.837666 70253 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot c7a52294 at applied index 17
I180911 21:58:09.837959 70253 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 48, log entries: 7, rate-limit: 2.0 MiB/sec, 1ms
I180911 21:58:09.838359 70513 storage/replica_raftstorage.go:787  [s2,r1/?:{-}] applying preemptive snapshot at index 17 (id=c7a52294, encoded size=8388, 1 rocksdb batches, 7 log entries)
I180911 21:58:09.839509 70513 storage/replica_raftstorage.go:793  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180911 21:58:09.839917 70253 storage/replica_command.go:812  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I180911 21:58:09.841749 70253 storage/replica.go:3765  [s1,r1/1:/M{in-ax},txn=59d33b55] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I180911 21:58:09.842846 70253 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n3 established
I180911 21:58:09.843705 70253 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot 8ad747aa at applied index 19
I180911 21:58:09.843976 70253 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 51, log entries: 9, rate-limit: 2.0 MiB/sec, 1ms
I180911 21:58:09.844383 70773 storage/replica_raftstorage.go:787  [s3,r1/?:{-}] applying preemptive snapshot at index 19 (id=8ad747aa, encoded size=9330, 1 rocksdb batches, 9 log entries)
I180911 21:58:09.845658 70773 storage/replica_raftstorage.go:793  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I180911 21:58:09.846153 70253 storage/replica_command.go:812  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=0]
I180911 21:58:09.847841 70542 rpc/nodedialer/nodedialer.go:92  connection to n1 established
I180911 21:58:09.849744 70253 storage/replica.go:3765  [s1,r1/1:/M{in-ax},txn=fe1f23f2] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I180911 21:58:09.851215 70253 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n4 established
I180911 21:58:09.852507 70253 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot 5beafd5e at applied index 21
I180911 21:58:09.852800 70253 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n4,s4):?: kv pairs: 54, log entries: 11, rate-limit: 2.0 MiB/sec, 2ms
I180911 21:58:09.853300 70526 storage/replica_raftstorage.go:787  [s4,r1/?:{-}] applying preemptive snapshot at index 21 (id=5beafd5e, encoded size=10337, 1 rocksdb batches, 11 log entries)
I180911 21:58:09.855075 70526 storage/replica_raftstorage.go:793  [s4,r1/?:/M{in-ax}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I180911 21:58:09.855708 70253 storage/replica_command.go:812  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n4,s4):4): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=0]
I180911 21:58:09.859164 70253 storage/replica.go:3765  [s1,r1/1:/M{in-ax},txn=9b75f1a3] proposing ADD_REPLICA((n4,s4):4): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3 (n4,s4):4] next=5
I180911 21:58:10.150995 70253 storage/replica_command.go:812  [s2,r1/2:/M{in-ax}] change replicas (REMOVE_REPLICA (n1,s1):1): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n4,s4):4, next=5, gen=0]
I180911 21:58:10.154496 70253 storage/replica.go:3765  [s2,r1/2:/M{in-ax},txn=83f1c13d] proposing REMOVE_REPLICA((n1,s1):1): updated=[(n4,s4):4 (n2,s2):2 (n3,s3):3] next=5
I180911 21:58:10.158361 70253 rpc/nodedialer/nodedialer.go:92  connection to n2 established
I180911 21:58:10.158416 70317 storage/store.go:2735  [replicaGC,s1,r1/1:/M{in-ax}] removing replica r1/1
I180911 21:58:10.158829 70778 storage/store.go:3827  [s1,r1/1:/M{in-ax}] added to replica GC queue (peer suggestion)
I180911 21:58:10.158853 70778 storage/store.go:3827  [s1,r1/1:/M{in-ax}] added to replica GC queue (peer suggestion)
I180911 21:58:10.159123 70317 storage/replica.go:871  [replicaGC,s1,r1/1:/M{in-ax}] removed 47 (42+5) keys in 1ms [clear=0ms commit=0ms]
I180911 21:58:11.635079 70872 gossip/gossip.go:1532  [n4] node has connected to cluster via gossip
I180911 21:58:11.666998 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n2 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:11.674319 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n2 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:11.698096 70753 gossip/gossip.go:942  [n3] gossip connectivity
  n2 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:11.700235 70625 gossip/gossip.go:1532  [n2] node has connected to cluster via gossip
I180911 21:58:11.776899 70872 gossip/gossip.go:942  [n4] gossip connectivity
  n2 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:11.810893 70531 gossip/gossip.go:1532  [n1] node has connected to cluster via gossip
I180911 21:58:12.077895 70753 gossip/gossip.go:1532  [n3] node has connected to cluster via gossip
W180911 21:58:15.164049 70625 gossip/gossip.go:1518  [n2] no incoming or outgoing connections
W180911 21:58:15.164120 70753 gossip/gossip.go:1518  [n3] no incoming or outgoing connections
W180911 21:58:15.164195 70872 gossip/gossip.go:1518  [n4] no incoming or outgoing connections
W180911 21:58:15.164252 70786 storage/raft_transport.go:583  while processing outgoing Raft queue to node 4: rpc error: code = Unavailable desc = transport is closing:
--- FAIL: test/TestReplicateRemovedNodeDisruptiveElection (5.520s)
client_raft_test.go:3197: did not get expected error
------- Stdout: -------
I180911 21:58:09.664432 70253 gossip/gossip.go:408  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:42405" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180911 21:58:09.666810 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n1 [sentinel];
 
W180911 21:58:09.671423 70253 gossip/gossip.go:1518  [n2] no incoming or outgoing connections
I180911 21:58:09.671803 70627 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:42405
I180911 21:58:09.674163 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n1 [sentinel];
 
I180911 21:58:09.674258 70253 gossip/gossip.go:408  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43957" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
W180911 21:58:09.696453 70253 gossip/gossip.go:1518  [n3] no incoming or outgoing connections
I180911 21:58:09.696551 70253 gossip/gossip.go:408  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:46753" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180911 21:58:09.697558 70758 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:42405
I180911 21:58:09.697702 70753 gossip/gossip.go:942  [n3] gossip connectivity
 
I180911 21:58:09.698267 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n1 [sentinel];
  n2 -> n1;
I180911 21:58:09.698772 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n1 [sentinel];
  n2 -> n1;
I180911 21:58:09.698995 70753 gossip/gossip.go:942  [n3] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I180911 21:58:09.699183 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
I180911 21:58:09.699267 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1;
W180911 21:58:09.775366 70253 gossip/gossip.go:1518  [n4] no incoming or outgoing connections
I180911 21:58:09.775494 70253 gossip/gossip.go:408  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:37093" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180911 21:58:09.776520 70309 gossip/client.go:129  [n4] started gossip client to 127.0.0.1:42405
I180911 21:58:09.776971 70872 gossip/gossip.go:942  [n4] gossip connectivity
 
I180911 21:58:09.780151 70872 gossip/gossip.go:942  [n4] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:09.781194 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:09.781748 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:09.781956 70753 gossip/gossip.go:942  [n3] gossip connectivity
  n1 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:09.837164 70253 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n2 established
I180911 21:58:09.837666 70253 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot c7a52294 at applied index 17
I180911 21:58:09.837959 70253 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 48, log entries: 7, rate-limit: 2.0 MiB/sec, 1ms
I180911 21:58:09.838359 70513 storage/replica_raftstorage.go:787  [s2,r1/?:{-}] applying preemptive snapshot at index 17 (id=c7a52294, encoded size=8388, 1 rocksdb batches, 7 log entries)
I180911 21:58:09.839509 70513 storage/replica_raftstorage.go:793  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180911 21:58:09.839917 70253 storage/replica_command.go:812  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I180911 21:58:09.841749 70253 storage/replica.go:3765  [s1,r1/1:/M{in-ax},txn=59d33b55] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I180911 21:58:09.842846 70253 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n3 established
I180911 21:58:09.843705 70253 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot 8ad747aa at applied index 19
I180911 21:58:09.843976 70253 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 51, log entries: 9, rate-limit: 2.0 MiB/sec, 1ms
I180911 21:58:09.844383 70773 storage/replica_raftstorage.go:787  [s3,r1/?:{-}] applying preemptive snapshot at index 19 (id=8ad747aa, encoded size=9330, 1 rocksdb batches, 9 log entries)
I180911 21:58:09.845658 70773 storage/replica_raftstorage.go:793  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I180911 21:58:09.846153 70253 storage/replica_command.go:812  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=0]
I180911 21:58:09.847841 70542 rpc/nodedialer/nodedialer.go:92  connection to n1 established
I180911 21:58:09.849744 70253 storage/replica.go:3765  [s1,r1/1:/M{in-ax},txn=fe1f23f2] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I180911 21:58:09.851215 70253 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n4 established
I180911 21:58:09.852507 70253 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot 5beafd5e at applied index 21
I180911 21:58:09.852800 70253 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n4,s4):?: kv pairs: 54, log entries: 11, rate-limit: 2.0 MiB/sec, 2ms
I180911 21:58:09.853300 70526 storage/replica_raftstorage.go:787  [s4,r1/?:{-}] applying preemptive snapshot at index 21 (id=5beafd5e, encoded size=10337, 1 rocksdb batches, 11 log entries)
I180911 21:58:09.855075 70526 storage/replica_raftstorage.go:793  [s4,r1/?:/M{in-ax}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I180911 21:58:09.855708 70253 storage/replica_command.go:812  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n4,s4):4): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=0]
I180911 21:58:09.859164 70253 storage/replica.go:3765  [s1,r1/1:/M{in-ax},txn=9b75f1a3] proposing ADD_REPLICA((n4,s4):4): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3 (n4,s4):4] next=5
I180911 21:58:10.150995 70253 storage/replica_command.go:812  [s2,r1/2:/M{in-ax}] change replicas (REMOVE_REPLICA (n1,s1):1): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n4,s4):4, next=5, gen=0]
I180911 21:58:10.154496 70253 storage/replica.go:3765  [s2,r1/2:/M{in-ax},txn=83f1c13d] proposing REMOVE_REPLICA((n1,s1):1): updated=[(n4,s4):4 (n2,s2):2 (n3,s3):3] next=5
I180911 21:58:10.158361 70253 rpc/nodedialer/nodedialer.go:92  connection to n2 established
I180911 21:58:10.158416 70317 storage/store.go:2735  [replicaGC,s1,r1/1:/M{in-ax}] removing replica r1/1
I180911 21:58:10.158829 70778 storage/store.go:3827  [s1,r1/1:/M{in-ax}] added to replica GC queue (peer suggestion)
I180911 21:58:10.158853 70778 storage/store.go:3827  [s1,r1/1:/M{in-ax}] added to replica GC queue (peer suggestion)
I180911 21:58:10.159123 70317 storage/replica.go:871  [replicaGC,s1,r1/1:/M{in-ax}] removed 47 (42+5) keys in 1ms [clear=0ms commit=0ms]
I180911 21:58:11.635079 70872 gossip/gossip.go:1532  [n4] node has connected to cluster via gossip
I180911 21:58:11.666998 70531 gossip/gossip.go:942  [n1] gossip connectivity
  n2 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:11.674319 70625 gossip/gossip.go:942  [n2] gossip connectivity
  n2 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:11.698096 70753 gossip/gossip.go:942  [n3] gossip connectivity
  n2 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:11.700235 70625 gossip/gossip.go:1532  [n2] node has connected to cluster via gossip
I180911 21:58:11.776899 70872 gossip/gossip.go:942  [n4] gossip connectivity
  n2 [sentinel];
  n2 -> n1; n3 -> n1; n4 -> n1;
I180911 21:58:11.810893 70531 gossip/gossip.go:1532  [n1] node has connected to cluster via gossip
I180911 21:58:12.077895 70753 gossip/gossip.go:1532  [n3] node has connected to cluster via gossip
W180911 21:58:15.164049 70625 gossip/gossip.go:1518  [n2] no incoming or outgoing connections
W180911 21:58:15.164120 70753 gossip/gossip.go:1518  [n3] no incoming or outgoing connections
W180911 21:58:15.164195 70872 gossip/gossip.go:1518  [n4] no incoming or outgoing connections
W180911 21:58:15.164252 70786 storage/raft_transport.go:583  while processing outgoing Raft queue to node 4: rpc error: code = Unavailable desc = transport is closing:

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -3172,8 +3172,9 @@ func TestReplicateRemovedNodeDisruptiveElection(t *testing.T) {
 	errChan := errorChannelTestHandler(make(chan *roachpb.Error, 1))
 	transport0.Listen(mtc.stores[0].StoreID(), errChan)
 
-	// Simulate an election triggered by the removed node.
-	transport0.SendAsync(&storage.RaftMessageRequest{
+	// Simulate an election triggered by the removed node. Try and try again
+	// until we're reasonably sure the message was sent.
+	for !transport0.SendAsync(&storage.RaftMessageRequest{
 		RangeID:     rangeID,
 		ToReplica:   replica1,
 		FromReplica: replica0,
@@ -3183,7 +3184,8 @@ func TestReplicateRemovedNodeDisruptiveElection(t *testing.T) {
 			Type: raftpb.MsgVote,
 			Term: term + 1,
 		},
-	})
+	}) {
+	}
 
 	// The receiver of this message should return an error.
 	select {

